lib/common/strings.c

```diff
@@ -599,24 +599,52 @@ crm_str_table_dup(GHashTable *old_table)
 
 /*!
  * \internal
- * \brief Add a word to a space-separated string list
+ * \brief Add a word to a string list of words
  *
- * \param[in,out] list  Pointer to beginning of list
- * \param[in]     word  Word to add to list
+ * \param[in,out] list       Pointer to current string list (may not be NULL)
+ * \param[in,out] len        If not NULL, must be set to length of \p list,
+ *                           and will be updated to new length of \p list
+ * \param[in]     word       String to add to \p list (\p list will be
+ *                           unchanged if this is NULL or the empty string)
+ * \param[in]     separator  String to separate words in \p list
+ *                           (a space will be used if this is NULL)
  *
- * \return (Potentially new) beginning of list
- * \note This dynamically reallocates list as needed.
+ * \note This dynamically reallocates \p list as needed. \p word may contain
+ *       \p separator, though that would be a bad idea if the string needs to be
+ *       parsed later.
  */
-char *
-pcmk__add_word(char *list, const char *word)
+void
+pcmk__add_separated_word(char **list, size_t *len, const char *word,
+                         const char *separator)
 {
-    if (word != NULL) {
-        size_t len = list? strlen(list) : 0;
+    size_t orig_len, new_len;
+
+    CRM_ASSERT(list != NULL);
 
-        list = pcmk__realloc(list, len + strlen(word) + 2); // 2 = space + EOS
-        sprintf(list + len, " %s", word);
+    if ((word == NULL) || (word[0] == '\0')) {
```

Comment:
You can use pcmk__str_empty for this.

Reply:
Doh! I'll also update cocci, it checks for word[0] == 0 but not '\0'

Reply:
cocci is the root canal of development tools, useful but painful